h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -1165,6 +1165,7 @@ public GLMWeights computeWeights(double y, double eta, double off, double w, GLM
           x.z = eta + (y-x.mu) *invSum * d2/(d*x.mu); // CHECKED-identity
         } else if (y==0 && x.mu > 0) {
           d = linkInvDeriv2(x.mu)*invSum-_theta*invSum*invSum*d2*d2; // CHECKED
+          //d = Math.min(1e10, Math.max(1e-10, d));
```

Comment:
Limiting the values of `d` here and in the previous if-branch (line 1163) significantly reduces the number of `NonSPDMatrixException` and `ArrayIndexOutOfBounds` (from `Gram.dropCols` mentioned in the beginning of this PR).

Limiting values of `d` on line 1168 seems to have significantly bigger effect (but that might just be due to extreme datasets are more often very zero inflated).

However, limiting `d` breaks the `hex.glm.GLMBasicTestNegativebinomial#testGradientLikelihoodTask` test (Hessian comparison).